src/build/build_step.go

```diff
@@ -329,8 +329,8 @@ func moveOutputs(state *core.BuildState, target *core.BuildTarget) ([]string, bo
 	changed := false
 	tmpDir := target.TmpDir()
 	outDir := target.OutDir()
-	for _, output := range target.Outputs() {
-		tmpOutput := path.Join(tmpDir, output)
+	for tmp, output := range target.GetOutPutsMapping(target.Outputs()) {
```

Comment:
Be careful with maps because they iterate in indeterminate order - I don't think it matters here but it would if we hashed them in the map's order.

Might be easier to just get both slices and iterate in parallel, then you don't need the mapping function.

Reply:
But what if the order of the different slices don't coordinate? like the tmpOutput doesn't map to the real output? I don't know if the order of the elements in the map actually matters, because it's the output for one specific target.

Reply:
As discussed, let's try having a function that maps a real output to the temp one, and skip having the map completely.